caddyconfig/caddyfile/dispenser.go

```diff
@@ -461,3 +461,17 @@ func (d *Dispenser) isNewLine() bool {
 	return d.tokens[d.cursor-1].File != d.tokens[d.cursor].File ||
 		d.tokens[d.cursor-1].Line+d.numLineBreaks(d.cursor-1) < d.tokens[d.cursor].Line
 }
+
+// isNextOnNewLine determines whether the current token is on a different
+// line (higher line number) than the next token. It handles imported
+// tokens correctly. If there isn't a next token, it returns true.
+func (d *Dispenser) isNextOnNewLine() bool {
+	if d.cursor < 0 {
+		return false
+	}
+	if d.cursor >= len(d.tokens)-1 {
+		return true
+	}
+	return d.tokens[d.cursor].File != d.tokens[d.cursor+1].File ||
+		d.tokens[d.cursor].Line+d.numLineBreaks(d.cursor) < d.tokens[d.cursor+1].Line
```

Comment:
This line broke my brain. Can you add a comment explaining the logic?

Reply:
Broke mine too. It's stolen from `isNewLine` but just shifted to look at `current < next` instead of `isNewLine` doing `prev < current`.

But yeah I think I'll break the `||` into a short-circuiting `if`, then clarify with comments

Reply:
Should be crystal clear now

Reply:
Brilliant!